src/Maker/MakeDto.php

```diff
@@ -0,0 +1,250 @@
+
+        $generator->writeChanges();
+        $manipulator = $this->createClassManipulator($DTOClassPath, $omitGettersSetters);
+        $mappedFields = $this->getMappedFieldsInEntity($metaData);
+
+        // Did we import assert annotations?
+        $assertionsImported = false;
+
+        foreach ($fields as $fieldName => $mapping) {
+            if (!\in_array($fieldName, $mappedFields)) {
+                continue;
+            }
+
+            $annotationReader = new AnnotationReader();
+
+            // Lookup classname for inherited properties
+            if (array_key_exists('declared', $mapping)) {
+                $fullClassName = $mapping['declared'];
+            } else {
+                $fullClassName = $boundClassDetails->getFullName();
+            }
+
+            // Property Annotations
+            $reflectionProperty = new \ReflectionProperty($fullClassName, $fieldName);
+            $propertyAnnotations = $annotationReader->getPropertyAnnotations($reflectionProperty);
+
+            $comments = [];
+
+            foreach ($propertyAnnotations as $annotation) {
+                // we want to copy the asserts, so look for their interface
+                if ($annotation instanceof Constraint) {
+                    $assertionsImported = true;
+                    $comments[] = $manipulator->buildAnnotationLine('@Assert\\'.(new \ReflectionClass($annotation))->getShortName(), (array) $annotation);
+                }
+            }
+
+            $manipulator->addEntityField($fieldName, $mapping, $comments);
+        }
+
+        $this->fileManager->dumpFile(
+            $DTOClassPath,
+            $manipulator->getSourceCode()
+        );
+
+        $this->writeSuccessMessage($io);
+
+        if (true === $assertionsImported) {
+            $io->note([
+                'The maker imported assertion annotations.',
+                'Consider removing them from the entity or make sure to keep them updated in both places.',
+            ]);
+        }
+
+        $io->text([
+            'Next: Create your form with this DTO and start using it:',
+            '$ php bin/console make:form '.$boundClassDetails->getShortName(),
+            '<fg=green>Enter fully qualified data class name to bind to the form:</>',
+            '> \\'.$dataClassNameDetails->getFullName(),
+            '',
+            'Find the documentation at <fg=yellow>https://symfony.com/doc/current/forms.html</>',
+        ]);
+    }
+
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+        $dependencies->requirePHP71();
+
+        $dependencies->addClassDependency(
+            Validation::class,
+            'validator',
+            // add as an optional dependency: the user *probably* wants validation
+            false
+        );
+    }
+
+    private function createClassManipulator(string $classPath, bool $omitGettersSetters = false): DTOClassSourceManipulator
+    {
+        return new DTOClassSourceManipulator(
```

Comment:
These DTO classes aren't meant to be updateable by this command, right? I mean, you run `make:dto` to create the class. But, you would not run it later to *update* the class after adding more fields to your entity, right? If so, I'm not sure we need to use the Parser magic to add the properties, getters & setters: we should just leverage passing variables into the template. The `ClassSourceManipulator` is useful when you need to update classes (because we can't do that with a simple template). But the complexity cost is high - I'm sure you know, because implementing this `DTOClassSourceManipulator` was probably quite challenging :).